jsonschema-hyperschema.xml

```diff
@@ -653,6 +662,91 @@
 
             </section>
 
+            <section title="hrefPointers" anchor="hrefPointers">
+                <t>
+                    The value of the "hrefPointers" link description property MUST be
+                    an object.  Each property value in the object MUST be a valid
+                    <xref target="I-D.luff-relative-json-pointer">Relative JSON Pointer</xref>,
+                    which is evaluated relative to the position in the instance from which
+                    <xref target="href">"href"</xref> template variable resolution would
+                    normally begin.
+                </t>
+                <t>
+                    For each property name in the object that matches a variable name in the
+                    "href" URI Template, the value of that property adjusts the starting position
+                    of variable resolution for that variable.  Properties which do not match
+                    "href" template variable names MUST be ignored.
+                </t>
+                <figure>
+                    <preamble>
+                        Consider the following schema for an n-ary tree using the path expansion
+                        URI Template syntax:
+                    </preamble>
+                    <artwork>
+<![CDATA[{
+    "type": "object",
+    "required": ["id"],
+    "properties": {
+        "id": {"type": "integer", "minimum": 0},
+        "children": {"type": "array", "items": {"$ref": "#"}}
+    },
+    "links": [
+        {
+            "rel": "self",
+            "href": "/trees{/root,parent,id}"
+            "hrefPointers": {
+                "root": "/id",
+                "parent": "2/id"
+            }
+        }
+    ]
+}]]>
+                    </artwork>
+                    <postamble>
+                        The self links here should always include the root node id,
+                        followed by an immediate parent id if one exists, and always ending
+                        with the id of the node that is the context of the self link.
+                        The context node's id resolves with the standard process and
+                        does not appear in "hrefPointers".  The root id is located
+                        using an absolute JSON Pointer, while the parent node
+                        uses a Relative JSON Pointer, going up two levels (one level up
+                        is the array of children, two levels up is the whole parent node).
+                    </postamble>
+                </figure>
+                <figure>
+                    <preamble>
+                        Given the following instance:
+                    </preamble>
+                    <artwork>
+<![CDATA[{
+  "id": 0,
+  "children": [ {
+    "id": 1,
+    "children": [ {
+      "id": 2,
+      "children": [ {
+        "id": 3
+} ] } ] } ] }]]>
```

Comment:
Yes, because it takes up absurd amounts of space otherwise.  I can do it closer the normal way, though.  I'll still leave the "[ {" / "} ]" pairs together because otherwise it really takes up too much vertical space.  But I'll put each closing pair on its own line, indented.

Reply:
Much better now!